src/TuxArchitectStores.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+//architecture OBJECT: stores will register their outputs to this object so that store inputs can lookup those outputs later
+var architecture = {};
+
+//ArchitectChain CONSTRUCTOR FUNCTION: produces architectChain instances which allow stores to chain methods for defining their inputs and outputs
+//@param storeToArchitect: store OBJECT that inputs and outputs will be registered to
+var ArchitectChain = function (storeToArchitect) {
+  this.storeToArchitect = storeToArchitect;
+};
+
+//prototype methods
+//_getRegisterId FUNCTION: internal function,for getting the store __registerId__ from either the passed in store or the store mapped to the passed in input
+//@param inputOrStore STRING: input string to be checked against the architecture OBJECT for the corresponding store [ALTERNATE OBJECT: if a store is passed in than lookup __registerId__ on the passed in store]
+ArchitectChain.prototype._getRegisterId = function (inputOrStore) {
+  var storeToWaitFor, storeRegistrationId;
+  //if inputOrStore is a string
+  if (typeof inputOrStore === 'string') {
+    //lookup corresponding store in the architecture object
+    storeToWaitFor = architecture[inputOrStore];
+    //if no store maps to this input throw error
+    if (!storeToWaitFor) {
+      throw new Error('store is waiting for an input: "' + inputOrStore + '" that no store outputs.  If this store needs no inputs than only call "itOutputs"');
+    }
+  } else {
+    //if inputOrStore is a store use that
+    storeToWaitFor = inputOrStore;
+  }
+  //return __registerId__ from storeToWaitFor
+  storeRegistrationId = storeToWaitFor.__registerId__;
+  if (storeRegistrationId) {
+    return storeRegistrationId;
+  } else {
+    //if store does not have __registerId__ than it has not been registered to the dispatcher so throw
+    throw new Error('store is waiting for a store that has not been registered to any actions');
```

Comment:
Recommend adding more information to this error. Specifically the inputOrStore string or information from the storeToWaitFor. Maybe the store name or something similar.